Documentation/Diagnostics/PH2155.md

```diff
@@ -34,20 +34,147 @@ Packages with unacceptable licenses, particularly copyleft licenses like GPL, ca
 <PackageReference Include="SomeGplPackage" Version="1.0.0" />
 ```
 
+### Diagnostic message
+
+```
+Package 'SomeGplPackage' version '1.0.0' has an unacceptable license 'GPL-3.0'. Consider adding to Allowed.Licenses.txt if license is acceptable.
+```
+
 ### Resolution options
 
+#### Option 1: Add to allowed licenses file
+
+```xml
+<!-- In your project file -->
+<ItemGroup>
+  <AdditionalFiles Include="Allowed.Licenses.txt" />
+</ItemGroup>
+```
+
 ```text
 # Allowed.Licenses.txt - Add acceptable licenses to this file
 MIT
 BSD-2-Clause
 BSD-3-Clause
 Apache-2.0
+# Commercial license for specific vendor
+CommercialLicense-Vendor-X
+# Allow GPL for this specific use case
+GPL-3.0
 ```
 
+#### Option 2: Find alternative package
+
+Replace the package with one that has a more permissive license:
+
+```xml
+<!-- Replace with alternative package -->
+<PackageReference Include="AlternativePackage" Version="2.0.0" />
+```
+
+#### Option 3: Remove the dependency
+
+If the package is not essential, remove it entirely:
+
+```xml
+<!-- Remove the problematic dependency -->
+<!-- <PackageReference Include="SomeGplPackage" Version="1.0.0" /> -->
+```
+
+## How the Analyzer Works
+
+The LicenseAnalyzer implements a sophisticated offline package license checking system that analyzes your project's NuGet dependencies without requiring network access:
+
+### Architecture
+
+1. **Project Discovery**: Uses `AnalyzerConfigOptionsProvider` to locate your project's `project.assets.json` file
```

Comment:
I believe the implementation foregoes using AnalyzerConfigOptionsProvider because of dependencies and .netstandard2.0 support.